Add Deno hosting documentation (#1896)
dotnet/docs-aspire: docs/community-toolkit/hosting-deno.md

## Hosting integration

To get started with the .NET Aspire Community Toolkit Deno extensions, install the [📦 CommunityToolkit.Aspire.Hosting.Deno](https://nuget.org/packages/CommunityToolkit.Aspire.Hosting.Deno) NuGet package in the AppHost project.

### [.NET CLI](#tab/dotnet-cli)

```dotnetcli
dotnet add package CommunityToolkit.Aspire.Hosting.Deno
```

### [PackageReference](#tab/package-reference)

```xml
<PackageReference Include="CommunityToolkit.Aspire.Hosting.Deno"
                  Version="*" />
```

---

For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).

## Example usage

The following sections detail various usages, from running Vite applications to using specific package managers.

### Run Deno apps

This integration extension adds support for running a Deno application defined in a script. Since [Deno is secure by default](https://docs.deno.com/runtime/fundamentals/security), permission flags must be specified in `permissionFlags` argument of `AddDenoApp`.

```csharp
var builder = DistributedApplication.CreateBuilder(args);

> builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["-E", "--allow-net"])

Reviewer: I don't know Deno very well, so maybe this is idiomatic as is, but as a newcomer I'd find `--allow-env` more helpful than `-E`.

They're equivalent according to the docs: https://docs.deno.com/runtime/fundamentals/security/#environment-variables

Author: What do you think of me showing both the long form and short form flags?

Reviewer 2: I saw 0433682 - but I'm not a fan of both examples in the code like that. I'd instead include text that says that the preceding code demonstrates passing the `permissionFlags` with their fully qualified switches, and that you could just as easily use the short-hand aliases and then list them there.

### [.NET CLI](#tab/dotnet-cli)

```dotnetcli
dotnet add package CommunityToolkit.Aspire.Hosting.Deno
```

### [PackageReference](#tab/package-reference)

```xml
<PackageReference Include="CommunityToolkit.Aspire.Hosting.Deno"
                  Version="*" />
```

---

For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).

## Example usage

The following sections detail various usages, from running Vite applications to using specific package managers.

### Run Deno apps

This integration extension adds support for running a Deno application defined in a script. Since [Deno is secure by default](https://docs.deno.com/runtime/fundamentals/security), permission flags must be specified in `permissionFlags` argument of `AddDenoApp`.

```csharp
var builder = DistributedApplication.CreateBuilder(args);

builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["--allow-env", "--allow-net"])
    .WithHttpEndpoint(env: "PORT")
    .WithEndpoint();

builder.Build().Run();
```

> *The preceding code uses the fully qualified switches. Short hand aliases can be used as well.*

Reviewer 2: ```suggestion
The preceding code uses the fully qualified switches. Alternatively, you can use the equivalent alias as well. For more information, see [Deno docs: Security and permissions](https://docs.deno.com/runtime/fundamentals/security/#permissions).
```

---

> title: Deno

Reviewer 2: ```suggestion
title: .NET Aspire Community Toolkit Deno hosting integration
```

description: Learn about the .NET Aspire Community Toolkit Deno hosting extensions package which provides functionality to run Deno applications and tasks.
ms.date: 10/25/2024
---

> # .NET Aspire Community Toolkit Deno Hosting

Reviewer 2: ```suggestion
# .NET Aspire Community Toolkit Deno hosting integration
```

### [.NET CLI](#tab/dotnet-cli)

```dotnetcli
dotnet add package CommunityToolkit.Aspire.Hosting.Deno
```

### [PackageReference](#tab/package-reference)

```xml
<PackageReference Include="CommunityToolkit.Aspire.Hosting.Deno"
                  Version="*" />
```

---

For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).

## Example usage

The following sections detail various usages, from running Vite applications to using specific package managers.

### Run Deno apps

This integration extension adds support for running a Deno application defined in a script. Since [Deno is secure by default](https://docs.deno.com/runtime/fundamentals/security), permission flags must be specified in `permissionFlags` argument of `AddDenoApp`.

```csharp
var builder = DistributedApplication.CreateBuilder(args);

builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["-E", "--allow-net"])
    .WithHttpEndpoint(env: "PORT")
    .WithEndpoint();

builder.Build().Run();
```

> 
> ---
> 

Reviewer 2: ```suggestion

```

### Run Deno tasks

> This integration extension adds support for running tasks that are either specified in a `package.json` or `deno.json`.

Reviewer 2: ```suggestion
This integration extension adds support for running tasks that are either specified in a _package.json_ or _deno.json_.
```

```csharp
var builder = DistributedApplication.CreateBuilder(args);

builder.AddDenoTask("vite-demo", taskName: "dev")
    .WithHttpEndpoint(env: "PORT")
    .WithEndpoint();

builder.Build().Run();
```

> 
> ---
> 

Reviewer 2: ```suggestion

```

> ### Deno Package Installation

Reviewer 2: ```suggestion
### Deno package installation
```

> This integration extension adds support for installing dependencies that utilizes `deno install` behind the scenes by simply using
> `WithDenoPackageInstallation`. **This only works with a `deno.lock` file present.**

Reviewer 2: ```suggestion
This integration extension adds support for installing dependencies that utilizes `deno install` behind the scenes by simply using
`WithDenoPackageInstallation`.

> [!NOTE]
> This API only works when a _deno.lock_ file present.
```

> 
> # [WithDenoPackageInstallation()](#tab/WithDenoPackageInstallation)
> 

Reviewer 2: ```suggestion

```

```csharp
var builder = DistributedApplication.CreateBuilder(args);

builder.AddDenoTask("vite-demo", taskName: "dev")
    .WithDenoPackageInstallation()
    .WithHttpEndpoint(env: "PORT")
    .WithEndpoint();
```

> 
> ---
> 

Reviewer 2: ```suggestion

```
